lib/printer.ts

```diff
@@ -1315,6 +1315,8 @@ function genericPrintNoParens(path: any, options: any, print: any) {
           ) {
             if (/\S/.test(child.value)) {
               return child.value.replace(/^\s+|\s+$/g, "");
+            } else if (/\n\n/.test(child.value)) {
```

Comment:
Fairly pedantic, but this won't work if there is any whitespace on the second empty line. You might consider making the regex something like `/\n[ \t]*\n/`